airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -150,8 +150,12 @@ public List<PartialAirbyteMessage> flushStates() {
       while (true) {
         final Long oldestState = stateIdQueue.peek();
         final boolean emptyQ = oldestState == null;
+        if (emptyQ) {
```

Comment:
nit: might be a bit more readable as
```java
if (stateIdQueue.isEmpty()) {
  break;
} 
final Long oldestState = stateIdQueue.peek();
```
(iirc that's the only time peek returns null)

Reply:
I like it! Now I have to run the tests..